Makefile

```diff
@@ -42,23 +65,47 @@ compile:
 	@$(REBAR) compile
 
 run:
+ifeq ($(REBAR_VSN),2)
 	erl -pa deps/*/ebin -pa ./ebin
+else
+	$(REBAR) shell
+endif
 
-eunit: compile
+eunit: $(TARGET_EUNIT_DEPS)
 	@$(REBAR) eunit skip_deps=true
 
-check: compile
+check: $(TARGET_CHECK_DEPS)
+check-eunit: $(TARGET_CHECK_EUNIT_DEPS)
+
+# Rebar3 knows how to run dialyzer
+dialyzer:
+	@$(REBAR) dialyzer
+
+# Manually run dialyzer when under rebar2
+dialyzer-2: compile
```

Comment:
+1 to this—it would make the Makefile more user-friendly and straightforward, I think.